src/identity/identity.service.ts

```diff
@@ -35,33 +38,42 @@ export class IdentityService {
     return deriveAddress({ public: publicKey }, secret, chainIdOf(address));
   }
 
-  async getDerivedIdentity(address: string, secret: string): Promise<object> {
+  async getDerivedIdentity(address: string, secret: string): Promise<DIDDocument> {
     const publicKey = await this.storage.getPublicKey(address);
 
     if (!publicKey) {
       return null;
     }
 
-    return this.asDidDocument(`${address}:derived:${secret}`, address, publicKey);
+    return this.asDidDocument(`${address}:derived:${secret}`, publicKey);
   }
 
-  asDidDocument(id: string, address: string, publicKey: string): object {
-    return {
+  async asDidDocument(id: string, publicKey: string): Promise<DIDDocument> {
+    const verificationMethods = await this.verificationMethodService.getMethodsFor(id);
+    const didDocument: DIDDocument = {
       '@context': 'https://www.w3.org/ns/did/v1',
-      'id': `did:lto:${id}`,
-      'verificationMethod': [{
-        id: `did:lto:${address}#key`,
+      id: `did:lto:${id}`,
+      verificationMethod: [{
+        id: `did:lto:${id}#key`,
```

Comment:
The `id` may contain `:derived:${secret}`, this should not be in the key of the verification method. It should just be the address. Therefore, pass the `address` as function param.

Reply:
done